h5py/h5fd.pyx

```diff
@@ -144,14 +144,14 @@ cdef herr_t H5FD_fileobj_close(H5FD_fileobj_t *f) except -1 with gil:
     stdlib_free(f)
     return 0
 
-cdef haddr_t H5FD_fileobj_get_eoa(const H5FD_fileobj_t *f, H5FD_mem_t type):
+cdef haddr_t H5FD_fileobj_get_eoa(const H5FD_fileobj_t *f, H5FD_mem_t type) noexcept nogil:
     return f.eoa
 
-cdef herr_t H5FD_fileobj_set_eoa(H5FD_fileobj_t *f, H5FD_mem_t type, haddr_t addr):
+cdef herr_t H5FD_fileobj_set_eoa(H5FD_fileobj_t *f, H5FD_mem_t type, haddr_t addr) noexcept nogil:
     f.eoa = addr
     return 0
 
-cdef haddr_t H5FD_fileobj_get_eof(const H5FD_fileobj_t *f, H5FD_mem_t type) except -1 with gil:  # HADDR_UNDEF
+cdef haddr_t H5FD_fileobj_get_eof(const H5FD_fileobj_t *f, H5FD_mem_t type) noexcept with gil:  # HADDR_UNDEF
     (<object>f.fileobj).seek(0, libc.stdio.SEEK_END)
     return (<object>f.fileobj).tell()
```

Comment:
This function can certainly fail, so it shouldn't be `noexcept`

Reply:
Ah ok, thank you. In what ways does it fail (where should I expect the failure from)? The access of the fileobj? From what I could tell `.tell()` on a python file-like object should never fail...oh but is `fileobj` a python file object or an HDF5/h5py file object? Seek 0 should never fail, right?

If/when it fails, what should it return? I guess this should be `except *` then?

Reply:
And I think it can be `except -1`, like it was before. The error value is actually `HADDR_UNDEF`, which is defined as the maximum possible uint64; -1 should be cast to this automatically.

Reply:
`get_eoa` is only returning `f.eoa` but I suppose that shouldn't be noexcept for the same reasons. `set_eoa` too?

Reply:
Oh but those are nogil so a normal python exception wouldn't work very well...right? Or does Cython figure that out when it exits the function?

Reply:
Those ones are OK, I think - although it looks like Python attribute access, `f.eoa` is actually accessing a field in a C struct (`ctypedef struct H5FD_fileobj_t` at the top of the file). So the get/set eoa functions don't touch any Python code, and they can therefore be noexcept and nogil.